esphome/components/am2320/am2320.cpp

```diff
@@ -0,0 +1,107 @@
+// Implementation based on:
+//  - ESPEasy: https://github.com/letscontrolit/ESPEasy/blob/mega/src/_P034_DHT12.ino
+//  - DHT12_sensor_library: https://github.com/xreef/DHT12_sensor_library/blob/master/DHT12.cpp
+//  - Arduino - AM2320: https://github.com/EngDial/AM2320/blob/master/src/AM2320.cpp
+
+#include "am2320.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace am2320 {
+
+static const char *TAG = "am2320";
+
+// ---=== Calc CRC16 ===---
+uint16_t crc_16(uint8_t *ptr, uint8_t length) {
+  uint16_t crc = 0xFFFF;
+  uint8_t i;
+  //------------------------------
+  while (length--) {
+    crc ^= *ptr++;
+    for (i = 0; i < 8; i++)
+      if ((crc & 0x01) != 0) {
+        crc >>= 1;
+        crc ^= 0xA001;
+      } else
+        crc >>= 1;
+  }
+  return crc;
+}
+
+void AM2320Component::update() {
+  uint8_t data[8];
+  data[0] = 0;
+  data[1] = 4;
+  if (!this->read_data_(data)) {
+    this->status_set_warning();
+    return;
+  }
+
+  float temperature = (((data[4] & 0x7F) << 8) + data[5]) / 10.0;
+  temperature = (data[4] & 0x80) ? -temperature : temperature;
+  float humidity = ((data[2] << 8) + data[3]) / 10.0;
+
+  ESP_LOGD(TAG, "Got temperature=%.2f°C humidity=%.2f%%", temperature, humidity);
```

Comment:
```suggestion
  ESP_LOGD(TAG, "Got temperature=%.1f°C humidity=%.1f%%", temperature, humidity);
```

both values only have 1 decimal place of accuracy.